docs/src/pages/guides/auth-protected/auth-protected.react.mdx

```diff
@@ -0,0 +1,414 @@
+
+```jsx
+// components/Login.js
+import { useEffect } from 'react';
+
+import { Authenticator, useAuthenticator, View } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+
+import { useNavigate, useLocation } from 'react-router';
+
+export function Login() {
+  const { route } = useAuthenticator((context) => [context.route]);
+  const location = useLocation();
+  const navigate = useNavigate();
+  let from = location.state?.from?.pathname || '/';
+
+  useEffect(() => {
+    if (route === 'authenticated') {
+      navigate(from, { replace: true });
+    }
+  }, [route, navigate, from]);
+
+  return (
+    <View className="auth-wrapper">
+      <Authenticator></Authenticator>
+    </View>
+  );
+}
+```
+
+Let's take a look at this line in more detail:
+
+```js
+const { route } = useAuthenticator((context) => [context.route]);
+```
+
+The [useAuthenticator](/components/authenticator#useauthenticator-hook) hook is used to access, modify, and update the
+Authenticator's auth state. In this example we destructure `useAuthenticator` and receive the `route`. The `route` can be used to tell if a route is `authenticated` or not.
+The `route` can also be set to `signIn`,`signOut`, or even `idle` depending on where the customer is during the authentication flow.
+
+> You may have noticed that we passed into the `useAuthenticator` a context callback function:
+> `(context) => [context.route]`
+> This is very important as it will help prevent unneccessary re-renders of the application. Excluding it may also cause errors!
+
+Let's take a look at the `useEffect` below:
+
+```js
+useEffect(() => {
+  if (route === 'authenticated') {
+    navigate(from, { replace: true });
+  }
+}, [route, navigate, from]);
+```
+
+When the `route` changes to `authenticated` the `useEffect` will trigger and the user will be navigated to the previous location.
+
+Alternatively, instead of using `route` this can also be accomplished using `Auth.currentAuthenticatedUser()`. This promise from the `aws-amplify` [library](https://docs.amplify.aws/lib/auth/manageusers/q/platform/js/)
+will return the `user` information back. If the user isn't logged in the promise will reject.
+
+## Adding in a RequireAuth Component
+
+Let's take a look at `RequireAuth` next. This component will be used to gate whether a route can be accessed or not. Routes that
+are **NOT** authenticated will be redirected back to `/login`. State will also be preserved.
+
+If the customer is logged in, the `route` will be set to `authenticated`, and the user can continue on.
+
+Create a new file called `RequireAuth.js` and add the code below:
+
+```jsx
+// RequireAuth.js
+import { useLocation, Navigate } from 'react-router-dom';
+import { useAuthenticator } from '@aws-amplify/ui-react';
+
+export function RequireAuth({ children }) {
+  const location = useLocation();
+  const { route } = useAuthenticator((context) => [context.route]);
+  if (route !== 'authenticated') {
```

Comment:
Would this line have the same problem on refresh where `route` is initially `idle` and prematurelly navigate to `/login` even if user is authenticated?

Reply:
Yes, it would. But the `/login` route has a check if they are `authenticated` or not. If they are authenticated it re-routes back. So the user wouldn't really notice any issue.

Maybe if they could try `Auth.currentAuthenticatedUser` if they didn't like the redirect back to the `/login` page. But that should only happen if they decide to refresh the page on a protected route.

Reply:
I'll add an additional note for the user explaining this scenario.

Reply:
7c6ee573f1e15b8407892dfd8552ea416e8cdf63

Reply:
Awesome, thanks!